### PR TITLE
Multi-arch enabled strip operations

### PR DESCRIPTION
**Pull request by TrevorTaoARM**
The current strip operations on Cilium related binaries
only work on building on x86_64, and actually the strip
operations commands are of the same name on both amd64
and arm64 platform which are defined by GNU:
for striping amd64 binary, it's
x86_64-linux-gnu-strip
for striping arm64 binary, it's
aarch64-linux-gnu-strip
The '/usr/bin/strip' is just a soft link to each for
each platform.

Here we just unify the 2 utilities on differrent
platforms (amd64, arm64), so that it can work on
both amd64 and arm64 platforms.

Signed-off-by: trevor tao <trevor.tao@arm.com>

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Fixes: #issue-number

```release-note
<!-- Enter the release note text here if needed or remove this section! -->
```

### CODE REVIEW

**Comment by aanm** [comment]
depends on https://github.com/cilium/cilium/pull/15110

**Review by aanm — CHANGES_REQUESTED** [review]
@TrevorTaoARM thanks for the changes, can you please follow the instructions here? https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

File: images/cilium/download-hubble.sh [hunk 1]
```diff
@@ -17,7 +17,6 @@ for arch in amd64 arm64 ; do
   printf "%s %s" "${hubble_sha256[${arch}]}" "/tmp/hubble-${arch}.tgz" | sha256sum -c
   mkdir -p "/out/linux/${arch}/bin"
   tar -C "/out/linux/${arch}/bin" -xf "/tmp/hubble-${arch}.tgz" hubble
+  [[ "${arch}" = "amd64" ]] && x86_64-linux-gnu-strip /out/linux/amd64/bin/hubble
+  [[ "${arch}" = "arm64" ]] && aarch64-linux-gnu-strip /out/linux/arm64/bin/hubble
```
Comment (aanm) [inline images/cilium/download-hubble.sh]: Also, this doesn't necessary needs to be in the for loop, it can be after, same for the other ones.

**Review by aanm — CHANGES_REQUESTED** [review]

File: images/cilium/download-hubble.sh (hunk 1 above)
Comment (TrevorTaoARM) [reply images/cilium/download-hubble.sh]: Yes, they are of the same function.
On the other side, we could think in this way: after we got 1 binary (hubble here), we do the strip operation then. 
So I think both of them are ok.

**Comment by TrevorTaoARM** [comment]
> @TrevorTaoARM thanks for the changes, can you please follow the instructions here? https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

OK, I would check and update it.

**Review by aanm — CHANGES_REQUESTED** [review]
@TrevorTaoARM the PR needs to be rebased first

File: images/builder/Dockerfile [hunk 2]
```diff
@@ -2,8 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ARG COMPILERS_IMAGE=quay.io/cilium/image-compilers:6ed96e618f68cd941b62b546fc8d161db0144d84@sha256:5245610a830b14972d5b287fc0318ac6ab0d420e4afc9b5db9032a04fd706d32
-ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:7f84e5f2c9027e75b271a460d83c086f29127d3a@sha256:a1d73d2840b4b075ade33bd8f968ca6db3d9bca9f3ed14f168d14526b3208cc0
-ARG TESTER_IMAGE=quay.io/cilium/image-tester:eb76f0d4d585946bb0267b4c4478cceed17bbd87@sha256:c9a598cdc4843e62ab4e430f36343ca06d5292506261f17e466e0f1cbb48ddbb
+ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:9922065e5fa27684f4e5848e450afc1b5d7c7f20
+ARG TESTER_IMAGE=docker.io/cilium/image-tester:70724309b859786e0a347605e407c5261f316eb0@sha256:89cc1f577d995021387871d3dbeb771b75ab4d70073d9bcbc42e532792719781
```
Comment (aanm) [inline images/builder/Dockerfile]: It looks like this line was accidentally modified.

**Review by aanm — CHANGES_REQUESTED** [review]

File: images/builder/Dockerfile [hunk 3]
```diff
@@ -2,8 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ARG COMPILERS_IMAGE=quay.io/cilium/image-compilers:6ed96e618f68cd941b62b546fc8d161db0144d84@sha256:5245610a830b14972d5b287fc0318ac6ab0d420e4afc9b5db9032a04fd706d32
-ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:7f84e5f2c9027e75b271a460d83c086f29127d3a@sha256:a1d73d2840b4b075ade33bd8f968ca6db3d9bca9f3ed14f168d14526b3208cc0
-ARG TESTER_IMAGE=quay.io/cilium/image-tester:eb76f0d4d585946bb0267b4c4478cceed17bbd87@sha256:c9a598cdc4843e62ab4e430f36343ca06d5292506261f17e466e0f1cbb48ddbb
+ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:9922065e5fa27684f4e5848e450afc1b5d7c7f20
+ARG TESTER_IMAGE=quay.io/cilium/image-tester:70724309b859786e0a347605e407c5261f316eb0@sha256:89cc1f577d995021387871d3dbeb771b75ab4d70073d9bcbc42e532792719781
```
Comment (aanm) [inline images/builder/Dockerfile]: This was also accidentally modified.

**Review by aanm — CHANGES_REQUESTED** [review]

File: images/builder/Dockerfile [hunk 4]
```diff
@@ -2,8 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ARG COMPILERS_IMAGE=quay.io/cilium/image-compilers:6ed96e618f68cd941b62b546fc8d161db0144d84@sha256:5245610a830b14972d5b287fc0318ac6ab0d420e4afc9b5db9032a04fd706d32
-ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:d1f58566ba3f12d7805d502cf137f8b6d27ac0fd@sha256:8dd763b845816f9e42122e38dc555fb6d85142132d0e060fe651c3fe53c83c44
-ARG TESTER_IMAGE=quay.io/cilium/image-tester:eb76f0d4d585946bb0267b4c4478cceed17bbd87@sha256:c9a598cdc4843e62ab4e430f36343ca06d5292506261f17e466e0f1cbb48ddbb
+ARG CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:9922065e5fa27684f4e5848e450afc1b5d7c7f20
+ARG TESTER_IMAGE=quay.io/cilium/image-tester:70724309b859786e0a347605e407c5261f316eb0@sha256:89cc1f577d995021387871d3dbeb771b75ab4d70073d9bcbc42e532792719781
```
Comment (aanm) [inline images/builder/Dockerfile]: @TrevorTaoARM can you remove this line change? Thanks

**Review by aanm — CHANGES_REQUESTED** [review]

File: images/builder/Dockerfile (hunk 4 above)
Comment (TrevorTaoARM) [reply images/builder/Dockerfile]: > @TrevorTaoARM can you remove this line change? Thanks
Yes, I just rebased the related Dockerfiles, and now it seems that only the 3 shell scripts are changed now. 
Did I forget any other steps to do?

**Review by aanm — CHANGES_REQUESTED** [review]
@TrevorTaoARM you need to follow these instructions https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

Please run `make -C images/ update-runtime-image`, add your changes into this commit and ping me back.

**Comment by aanm** [comment]
> > @TrevorTaoARM you need to follow these instructions https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images
> > Please run `make -C images/ update-runtime-image`, add your changes into this commit and ping me back.
> > @aanm I added all the modified files after running `make -C images/ update-runtime-image`. Please check if it's ok.

@TrevorTaoARM thank you, please run `make -C images/ update-runtime-image update-builder-image`, re-add the changes and push them again.

**Comment by TrevorTaoARM** [comment]
> > > @TrevorTaoARM you need to follow these instructions https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images
> > > Please run `make -C images/ update-runtime-image`, add your changes into this commit and ping me back.
> > > @aanm I added all the modified files after running `make -C images/ update-runtime-image`. Please check if it's ok.
> 
> @TrevorTaoARM thank you, please run `make -C images/ update-runtime-image update-builder-image`, re-add the changes and push them again.

@aanm Sorry, I did the steps in https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images
again, so the whole things may need to be rewinded since I have not understood the steps previously.
Please tell me how to do now if anything wrong.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by nbusseneau — APPROVED** [review]
LGTM, but seeing the failures in https://jenkins.cilium.io/job/Cilium-PR-K8s-1.21-kernel-4.9/45/#showFailuresLink perhaps we may want to rebase this on `master` (these 4 failures are regressions that have been fixed since then).

**Comment by TrevorTaoARM** [comment]
> LGTM, but seeing the failures in https://jenkins.cilium.io/job/Cilium-PR-K8s-1.21-kernel-4.9/45/#showFailuresLink perhaps we may want to rebase this on `master` (these 4 failures are regressions that have been fixed since then).

@nbusseneau As what you said, I rebased this on 'master'. Please check if it's ok.

**Comment by nbusseneau** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by nbusseneau** [comment]
test-gke

**Comment by joestringer** [comment]
@TrevorTaoARM @aanm thanks again for the PR, as a quick reminder for future PRs, please ensure that you either update the release note in the description or remove this section as these end up in the final release notes for the upcoming release:

```
<!-- Enter the release note text here if needed or remove this section! -->
```

**Pull request merged**